nightfall/detection_rules.py

```diff
@@ -0,0 +1,251 @@
+        uuid (str or None): Create an instance of a detector by using an existing detector's UUID.
+        display_name (str or None): A display name for this detector.
+        context_rules (List[ContextRule] or None): The context rules to use to customize the behavior of this detector.
+        exclusion_rules (List[ExclusionRule] or None): The exclusion rules to use to customize the behavior of this
+            detector.
+        redaction_config (RedactionConfig or None): Sets the redaction configuration to-be-applied to this detector.
+            This configuration is currently only supported for scanning plaintext, not for file scanning.
+    """
+    min_confidence: Confidence
+    min_num_findings: int = 1
+    nightfall_detector: str = None
+    regex: Regex = None
+    word_list: WordList = None
+    uuid: str = None
+    display_name: str = None
+    context_rules: List[ContextRule] = None
+    exclusion_rules: List[ExclusionRule] = None
+    redaction_config: RedactionConfig = None
+
+    def __post_init__(self):
+        if [self.nightfall_detector, self.regex, self.word_list, self.uuid].count(None) != 3:
+            raise NightfallUserError("need one of nightfall_detector, regex, word_list, or uuid", 40001)
+
+    def as_dict(self):
+        result = {"minConfidence": self.min_confidence.value, "minNumFindings": self.min_num_findings}
+        if self.nightfall_detector:
+            result["nightfallDetector"] = self.nightfall_detector
+            result["detectorType"] = "NIGHTFALL_DETECTOR"
+        if self.regex:
+            result["regex"] = self.regex.as_dict()
+            result["detectorType"] = "REGEX"
+        if self.word_list:
+            result["wordList"] = self.word_list.as_dict()
+            result["detectorType"] = "WORD_LIST"
+        if self.uuid:
+            result["uuid"] = self.uuid
+        if self.display_name:
+            result["displayName"] = self.display_name
+        if self.context_rules:
+            result["contextRules"] = [c.as_dict() for c in self.context_rules]
+        if self.exclusion_rules:
+            result["exclusionRules"] = [e.as_dict() for e in self.exclusion_rules]
+        if self.redaction_config:
+            result["redactionConfig"] = self.redaction_config.as_dict()
+        return result
+
+
+class LogicalOp(Enum):
+    """ A modifier that is used to decide when a finding should be surfaced in the context of a detection rule.
+    - When ALL is specified, all detectors in a detection rule must trigger a match in order for the finding to be
+      reported. This is the equivalent of a logical "AND" operator.
+    - When ANY is specified, only one of the detectors in a detection rule must trigger a match in order for the finding
+      to be reported. This is the equivalent of a logical "OR" operator.
+    """
+    ANY = "ANY"
+    ALL = "ALL"
+
+
+@dataclass
+class DetectionRule:
+    """An object that contains a set of detectors to be used when scanning content.
```

Comment:
`DetectionRules` have an optional `name` property. Was this excluded on purpose?